### DISCUSSION TITLE
ngOnChanges vs ngDoCheck: Misunderstanding When the ngDoCheck Gets Triggered

### DISCUSSION THREAD

**Discussion by prishnah** [clovercommunity/resco#5] [Q&A] [answered]
I have used OnPush strategy for my component and no bindings have changed, but the ngDoChecklifecycle hook is still triggered. Is the strategy not working?

**Comment by davatas** [comment]
Yes, I bet that you've tumbled across this question, maybe put into different words, on various developer forums.

**Comment by albertcodes** [comment, marked as answer]
Using the _ngDoCheck_ lifecycle hook once you run the change detection process.
The **onPush** strategy for detecting changes in Angular does work: the hook is triggered by the design itself.
